enterprise_gateway/services/processproxies/processproxy.py

```diff
@@ -795,7 +795,7 @@ def _update_connection(self, connect_info):
 
         # If there's a response-socket, close it since its no longer needed.
         if self.response_socket:
-            self.response_socket.shutdown(SHUT_RDWR)
+            self.response_socket.shutdown_kernel(SHUT_RDWR)
```

Comment:
Hmm - I think you got overzealous on your replacement.  I doubt sockets have `shutdown_kernel` method. :smile: